homeassistant/components/poolstation/__init__.py

```diff
@@ -0,0 +1,75 @@
+"""The Poolstation integration."""
+from datetime import timedelta
+import logging
+
+import aiohttp
+from pypoolstation import Account, Pool
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.aiohttp_client import async_create_clientsession
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import COORDINATORS, DEVICES, DOMAIN, TOKEN
+
+PLATFORMS = ["sensor"]
+
+_LOGGER = logging.getLogger(__name__)
+
+UPDATE_INTERVAL = timedelta(seconds=30)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Poolstation from a config entry."""
+    session = async_create_clientsession(hass, cookie_jar=aiohttp.DummyCookieJar)
+    account = Account(session, token=entry.data[TOKEN])
+
+    try:
+        pools = await Pool.all(session, account=account)
```

Comment:
To me it seems a little bit odd when a method has the same name as a Python built-in function. https://docs.python.org/3/library/functions.html
Especially when it doesn't yield similar results.

Reply:
What method do you suggest? I know nothing about python, but knowing ruby `ClassName.all` felt quite natural to return all the instances of that given class.

Reply:
Im not a seasoned Python Developer too (just doing this for fun) so I'm not sure it is even a concern for others. 
But when you return all instances (there may even be a built-in method for that) I'd suggest `.instances`

Reply:
I would add `get_all_pools()` to the Account class (or rename `Pool.all()` to `Pool.get_all_pools_from_account()`).

Reply:
I went with `get_all_pools()`.